src/ripple/app/tx/impl/CreateOffer.cpp

```diff
@@ -629,6 +626,406 @@ CreateOffer::cross (
     }
 }
 
+std::pair<TER, Amounts>
+CreateOffer::crossV2 (
+    PaymentSandbox& psb,
+    PaymentSandbox& psbCancel,
+    Amounts const& takerAmount)
+{
+    try
+    {
+        // If the taker is unfunded before we begin crossing there's nothing
+        // to do - just return an error.
+        //
+        // We check this in preclaim, but when selling XRP charged fees can
+        // cause a user's available balance to go to 0 (by causing it to dip
+        // below the reserve) so we check this case again.
+        STAmount const inStartBalance = accountFunds (
+            psb, account_, takerAmount.in, fhZERO_IF_FROZEN, j_);
+        if (inStartBalance.signum() <= 0)
```

Comment:
I'd prefer the more "obvious" check here: `if (inStartBalance <= zero)` which clearly expresses the code's intent.